packages/compass-crud/src/components/bulk-update-dialog.tsx

```diff
@@ -0,0 +1,306 @@
+
+import type { BSONObject } from '../stores/crud-store';
+import type { UpdatePreview } from 'mongodb-data-service';
+
+const columnsStyles = css({
+  marginTop: spacing[4],
+  display: 'grid',
+  width: '100%',
+  gap: spacing[4],
+  gridTemplateColumns: 'repeat(2, minmax(0, 1fr))',
+});
+
+const queryStyles = css({
+  display: 'flex',
+  gap: spacing[4],
+  flexDirection: 'column',
+});
+
+const queryFieldStyles = css({});
+
+const updateFieldStyles = css({
+  flex: 1,
+});
+
+const descriptionStyles = css({
+  marginBottom: spacing[2],
+});
+
+const previewStyles = css({
+  contain: 'size',
+  overflow: 'scroll',
+});
+
+const previewDescriptionStyles = css({
+  display: 'inline',
+});
+
+const codeContainerStyles = css({
+  paddingTop: spacing[2],
+  paddingBottom: spacing[2],
+});
+
+const codeDarkContainerStyles = css({
+  background: palette.gray.dark4,
+});
+
+const codeLightContainerStyles = css({
+  backgroundColor: palette.gray.light3,
+});
+
+const multilineContainerStyles = css({
+  maxHeight: spacing[4] * 20,
+});
+
+// We use custom color here so need to disable default one that we use
+// everywhere else
+const codeEditorStyles = css({
+  background: 'transparent !important',
+  '& .cm-gutters': {
+    background: 'transparent !important',
+  },
+  '& .cm-editor': {
+    background: 'transparent !important',
+  },
+});
+
+const bannerContainerStyles = css({
+  // don't jump when an error appears
+  minHeight: spacing[4] * 2 + 2,
+  overflow: 'hidden',
+});
+
+const bannerStyles = css({
+  flex: 'none',
+  marginTop: spacing[2],
+  marginLeft: spacing[2],
+  marginRight: spacing[2],
+  textAlign: 'left',
+});
+
+const updatePreviewStyles = css({
+  display: 'flex',
+  flexDirection: 'column',
+  gap: spacing[3],
+});
+
+export type BulkUpdateDialogProps = {
+  isOpen: boolean;
+  ns: string;
+  filter: BSONObject;
+  count?: number;
+  updateText: string;
+  preview: UpdatePreview;
+  syntaxError?: Error & { loc?: { index: number } };
+  serverError?: Error;
+  closeBulkUpdateDialog: () => void;
+  updateBulkUpdatePreview: (updateText: string) => void;
+  runBulkUpdate: () => void;
+};
+
+type QueryLabelProps = {
+  tooltip: string;
+  label: string;
+};
+
+const queryLabelStyles = css({
+  display: 'flex',
+  gap: spacing[2],
+  alignItems: 'center',
+});
+
+const QueryLabel: React.FunctionComponent<QueryLabelProps> = ({
+  tooltip,
+  label,
+}) => {
+  return (
+    <div className={queryLabelStyles}>
+      <Label htmlFor="template-dropdown">{label}</Label>
+      <InfoSprinkle align="right">{tooltip}</InfoSprinkle>
+    </div>
+  );
+};
+
+export default function BulkUpdateDialog({
+  isOpen,
+  ns,
+  filter,
+  count,
+  updateText,
+  preview,
+  syntaxError,
+  serverError,
+  closeBulkUpdateDialog,
+  updateBulkUpdatePreview,
+  runBulkUpdate,
+}: BulkUpdateDialogProps) {
+  const darkMode = useDarkMode();
+
+  const [text, setText] = useState(updateText);
```

Comment:
Doesn't look like we reset this when modal is re-opened, is this expected?

Reply:
Yup.

Reply:
Maybe worth leaving a comment that it's on purposes, we have a bunch of modal forms where we reset forms on modal open, I'm worried we can loose this behavior by accident